libgadget/run.c

```diff
@@ -359,11 +363,11 @@ void compute_accelerations(int is_PM, int FirstStep, int GasEnabled, int HybridN
      * this timestep to GravPM. Note initially both
      * are zero and so the tree is opened maximally
      * on the first timestep.*/
-    grav_short_tree(tree);
-    /* TreeUseBH > 1 means use the BH criterion on the initial timestep only,
-     * avoiding the fully open O(N^2) case.*/
-    if(All.TreeUseBH > 1)
-        All.TreeUseBH = 0;
+    const int NeutrinoTracer =  All.HybridNeutrinosOn && (All.Time <= All.HybridNuPartTime);
+    const double rho0 = All.CP.Omega0 * 3 * All.CP.Hubble * All.CP.Hubble / (8 * M_PI * All.G);
+
+    if(All.TreeGravOn)
+        grav_short_tree(act, pm, tree, All.G, rho0, NeutrinoTracer, All.FastParticleType);
```

Comment:
grav_short_tree requires pm.. That's a bit surprising, but understandable..

Reply:
I thought about it and it seems actually cleaner: it indicates that the short-range gravity depends on how the PM mesh is set up